packages/salesforcedx-utils-vscode/src/commands/commandEventStream.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as vscode from 'vscode';
+
+export enum CommandEventType {
+  START = 'start',
+  END = 'end',
+  EXIT_CODE = 'exitCode',
+  DATA = 'data',
+  ERROR = 'error'
+}
+
+export type CommandEventStart = {
+  type: CommandEventType.START;
+  commandId: string;
+};
+
+export type CommandEventEnd = {
+  type: CommandEventType.END;
+  commandId: string;
+};
+
+export type CommandEventExitCode = {
+  type: CommandEventType.EXIT_CODE;
+  exitCode: number;
+};
+
+export type CommandEventData = {
+  type: CommandEventType.DATA;
+  data: any;
+};
+
+export type CommandEventError = {
+  type: CommandEventType.ERROR;
+  error: string;
+};
+
+export type CommandEvent = CommandEventStart | CommandEventEnd | CommandEventExitCode | CommandEventData | CommandEventError;
+
+export class CommandEventStream {
+  private static instance: CommandEventStream;
+  private readonly eventEmitter = new vscode.EventEmitter<CommandEvent>();
```

Comment:
should this be assigned in the constructor for testability?

Reply:
TS complains somewhere about the member not being initialized if I do that, when I write the tests I'll see if I can refactor to make everything happy.